scripts/src/javatests/org/oppia/android/scripts/maven/GenerateLicenseTextsTest.kt

```diff
@@ -0,0 +1,576 @@
+    )
+    val dependencyList = listOf<MavenDependency>(
+      getMavenDependency(
+        artifactName = "artifact.name:1",
+        artifactVersion = "2.0.1",
+        licenseList = listOf(scrapableLinkLicense)
+      ),
+      getMavenDependency(
+        artifactName = "artifact.name:2",
+        artifactVersion = "4.1.1",
+        licenseList = listOf(directLinkOnlyLicense, extractedCopyLinkLicense)
+      ),
+      getMavenDependency(
+        artifactName = "artifact.name:3",
+        artifactVersion = "3.1.1",
+        licenseList = listOf(scrapableLinkLicense)
+      )
+    )
+    val mavenDependencyList = getMavenDependencyList(dependencyList)
+
+    val pbFile = tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+    val xmlFile = tempFolder.newFile("values/third_party_dependencies.xml")
+    mavenDependencyList.writeTo(pbFile.outputStream())
+
+    GenerateLicenseTexts(mockLicenseFetcher).main(
+      arrayOf(
+        "${tempFolder.root}/values",
+        "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+      )
+    )
+    assertThat(outContent.toString()).contains(SCRIPT_PASSED_INDICATOR)
+    val xmlContent = xmlFile.inputStream().bufferedReader().use { it.readText() }
+    val dependencyNamesList = retrieveListOfStrings(xmlContent, "third_party_dependency_name_")
+    val dependencyVersionsList =
+      retrieveListOfStrings(xmlContent, "third_party_dependency_version_")
+    val licenseTextsList = retrieveListOfStrings(xmlContent, "license_text_")
+    val licenseNamesList = retrieveListOfStrings(xmlContent, "license_name_")
+
+    verifyList(
+      itemList = dependencyNamesList,
+      expectedSize = 3,
+      expectedList = listOf(
+        "artifact.name:1",
+        "artifact.name:2",
+        "artifact.name:3"
+      )
+    )
+    verifyList(
+      itemList = dependencyVersionsList,
+      expectedSize = 3,
+      expectedList = listOf(
+        "2.0.1",
+        "4.1.1",
+        "3.1.1"
+      )
+    )
+    verifyList(
+      itemList = licenseTextsList,
+      expectedSize = 3,
+      expectedList = listOf(
+        """
+        "                        Apache License
+        "License" shall mean the terms and conditions for use, reproduction,
+        and distribution as defined by Sections 1 through 9 of this document."
+        """.trimIndent(),
+        "\"$DIRECT_LINK_ONLY\"",
+        """
+        "Copyright <YEAR> <COPYRIGHT HOLDER>
+
+        Redistribution and use in source and binary forms, with or without modification, are 
+        permitted provided that the following conditions are met:"
+        """.trimIndent()
+      )
+    )
+    verifyList(
+      itemList = licenseNamesList,
+      expectedSize = 3,
+      expectedList = listOf(
+        "Apache License",
+        "Android Terms of Service",
+        "BSD License"
+      )
+    )
+
+    val dependencyNamesArray = retrieveArray(
+      xmlContent,
+      "third_party_dependency_names_array",
```

Comment:
add named argument for this and other similar cases

Reply:
Done.